Homework № 6 "Unit testing". Try1.

IgushkinV opened the pull request.

IgushkinV committed:
Homework № 6 "Unit testing". Try1.

IgushkinV committed:
2.22.2 plugin version.

IgushkinV committed:
Delete plugin dependency.

tlt-java-courses commented on src/test/java/Igushkin/Lesson2/CacheTest.java:
```diff
@@ -0,0 +1,73 @@
+class CacheTest {
+    private int capacityOfCache = 1;
+    private int lastIndex;
+    int testIndex = 0;
```
Зачем здесь capacityOfCache, lastIndex, testIndex  если они нигде не используются?

IgushkinV replied:
Забыл убрать, изначально хотел использовать. Убрал.

tlt-java-courses commented on src/test/java/Igushkin/Lesson2/CacheTest.java:
```diff
@@ -0,0 +1,73 @@
+
+        cache.add("test",0);
+        boolean actualResult = cache.isPresent(testElement);
+        assertEquals(expectedResult, actualResult);
```
Для проверки на true/false лучше использовать соотвествующие ассерты:
assertTrue / assertFalse

IgushkinV replied:
Переписал.

tlt-java-courses commented on src/test/java/Igushkin/Lesson2/CacheTest.java:
```diff
@@ -0,0 +1,73 @@
+
+    @Test
+    void isPresentByIndexReturnsTrueWhenElementIsPresent() {
+        CacheElement<String> element = new CacheElement<>("test", 0);
```
Нигде не используется

IgushkinV replied:
Убрал.

tlt-java-courses commented on src/test/java/Igushkin/Lesson2/CacheTest.java:
```diff
@@ -0,0 +1,73 @@
+    void clearReturnsCacheWithOnlyNulls() {
+        cache.add(testString, 0);
+        cache.clear();
+        assertEquals(null, cache.get(0));
```
Используй соотвествующий ассерт: assertNull

IgushkinV replied:
Исправил.

tlt-java-courses commented on src/test/java/Igushkin/Lesson2/CacheTest.java:
```diff
@@ -0,0 +1,73 @@
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CacheTest {
```
Не нашел теста на ситуацию когда при добавлении происходит сдвиг массива

IgushkinV replied:
Написал тест для этой ситуации.

tlt-java-courses commented on src/test/java/Igushkin/Lesson2/StorageTest.java:
```diff
@@ -0,0 +1,103 @@
+    @Test
+    void addReallyAddingElementToStorage() throws NegativeIndexException {
+        testStorage.add(testString);
+        assertTrue(testStorage.getLast() != null);
```
используй assertNotNull

IgushkinV replied:
Исправил.

tlt-java-courses commented on src/test/java/Igushkin/Lesson2/StorageTest.java:
```diff
@@ -0,0 +1,103 @@
+    void clearReturnsStorageWithOnlyNulls() throws NegativeIndexException {
+        testStorage.add(testString);
+        testStorage.clear();
+        assertEquals(null, testStorage.get(0));
```
assertNull

IgushkinV replied:
Исправил.

IgushkinV committed:
Исправление замечаний к ДЗ.
Удалил ненужные поля.
Использовал assertTrue и assertNull, где это было необходимо.
Добавил проверку корректного элемента после перемещения элементов в кэше.

IgushkinV committed:
Исправление замечаний к ДЗ.
Исправление замечаний, неисправленных в первом коммите.

IgushkinV committed:
Добавление теста на случай переполнения кэша и передвижения элементов кэша в связи с этим.
